content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,228 @@
+- From there, finally be redirected into the target pod’s network namespace (and processed by any rules there).
+
+In Kubernetes, the [CRI](https://kubernetes.io/docs/concepts/architecture/cri/) (container *runtime* interface) is responsible for talking to the Linux kernel, creating network namespaces
+for new pods, and starting processes within them. The CRI then invokes the [CNI](https://github.com/containernetworking/cni) (container *networking* interface),
+which is responsible for wiring up the networking rules in the various Linux network namespaces, so that packets leaving and
+entering the new pod can get where they’re supposed to go. It doesn’t matter much what topology or mechanism the CNI uses to
+accomplish this - as long as packets get where they’re supposed to be, Kubernetes works and everyone is happy.
+
+### Istio ambient traffic redirection: why did we drop the previous model?
+
+In Istio ambient, every node has a minimum of two containers running as Kubernetes DaemonSet:
+- An efficient ztunnel which handles mesh traffic proxying duties, and L4 policy enforcement.
+- A CNI node agent that handles enrolling new and existing pods into the ambient mesh.
+
+In the previous ambient model, this is how mesh enrollment worked:
+- A Kubernetes pod (existing or newly-started), with its namespace labeled with `istio.io/dataplane-mode=enabled` indicating it should
+be included in the ambient mesh, is detected by the istio-cni node agent.
+- The istio-cni node agent then establishes network redirection rules in the top-level node network namespace, such that
+packets entering or leaving the enrolled pod would be intercepted and redirected to that node’s ztunnel on the relevant
+proxy ports (15008, 15006, or 15001).
+
+This meant that for a packet created by a mesh-enrolled pod, that packet would leave that source pod, enter the node’s
+top-level network namespace, and then be intercepted and redirected to that node’s ztunnel (running in its own network
+namespace) for proxying to the destination pod, with the return trip being similar.
+
+This model worked well enough as a placeholder for the initial Ambient launch, but as mentioned, it has a fundamental
+problem - there are many CNI implementations, and in Linux there are many fundamentally different and incompatible ways
+in which you can configure how packets get from one network namespace to another. You can use tunnels, overlay networks,
+you can go through the host network namespace, or you can bypass it. You can go through the Linux user space networking stack,
+or you can skip it and shuttle packets back and forth in the kernel space stack, etc etc. And for every possible approach,
+there’s probably a CNI implementation out there that makes use of it.
+
+Which meant that with the previous redirection approach, there were lots of CNI implementations Ambient simply wouldn’t
```

Comment:
multiple sentences.

"... Ambient wouldn't work with. Given its reliance on node network ... any CNI that didn't ...".

Reply:
addressed